client/src/components/assessments/AssessmentResultsTable.js

```diff
@@ -0,0 +1,325 @@
+import AggregationWidget from "components/AggregationWidgets"
+import AddAssessmentModal from "components/assessments/AddAssessmentModal"
+import { ReadonlyCustomFields } from "components/CustomFields"
+import Fieldset from "components/Fieldset"
+import { Formik } from "formik"
+import LinkTo from "components/LinkTo"
+import _isEmpty from "lodash/isEmpty"
+import PropTypes from "prop-types"
+import React, { useState } from "react"
+import { Button, Table } from "react-bootstrap"
+import "components/assessments/AssessmentResultsTable.css"
+
+/* The AssessmentResultsTable component displays the results of two types of
+ * assessments made on a given entity and subentities:
+ * - aggregation of the measurements made on the entity/subentities when
+ *   working on them in relation to another type of entity (example:
+ *   assessments made on tasks, while filling  report related to the tasks);
+ *   the definition of these assessments is to be found in
+ *   entity.customFields.assessmentDefinition
+ * - display of the last monthly assessment made on the entity/subentities
+ *   as a conclusion about the given period of time;
+ *   the definition of these assessments is to be found in
+ *   entity.periodAssessmentConfig()
+ */
+
+const AssessmentsTableHeader = ({ periods }) => (
+  <thead>
+    <tr key="periods">
+      <>
+        {periods.map(period => (
+          <th key={period.start}>{period.start.format("MMM-YYYY")}</th>
+        ))}
+      </>
+    </tr>
+  </thead>
+)
+AssessmentsTableHeader.propTypes = {
+  periods: PropTypes.array
+}
+
+const MeasurementRow = ({
+  measurementKey,
+  measurementDef,
+  entity,
+  assessmentPeriods
+}) => {
+  const aggWidgetProps = {
+    widget: measurementDef.aggregation?.widget || measurementDef.widget,
+    aggregationType: measurementDef.aggregation?.aggregationType,
+    vertical: true
+  }
+  const widgetLayoutConfig = Object.without(
+    measurementDef,
+    "aggregation",
+    "type",
+    "typeError",
+    "placeholder",
+    "helpText",
+    "validations",
+    "visibleWhen",
+    "objectFields"
+  )
```

Comment:
According to the schema, it can have the fields:
```type, typeError, label, helpText, visibleWhen, style, placeholder, componentClass, validations, aggregation, choices, addButtonLabel, objectLabel, objectFields, widget, levels```
Wouldn't it be clearer to list the fields you actually *do* want instead?

Reply:
But lots of the other properties might not be available.

Reply:
OK. To me it seems prone to break with such a long list of exceptions, so perhaps it's better to have the widget props in a separate sub-element `measurementDef.widgetProps.<props-here>` so they're all in once place.

Reply:
I will not add a widgetProps sub-element to the custom fields, as this would also need migration of current JSON values in the database and I am not sure the splitting is always needed like that.
But I have added a function which retrieves the props and is now being reused in 3 places. This way it is a bit less error prone and the code is more readable.